components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -5636,13 +5665,14 @@ private void populateBasicAttributes(LocalClaim mappedLocalClaim, AbstractAttrib
                 attribute.setType(SCIMDefinitions.DataType.STRING);
             }
 
-        } else if (isBooleanAttribute(attribute.getName())) {
+        } else if (isBooleanInAdditionalProp || (isBooleanAttribute(attribute.getName()) && !isCustomSchemaAttr)) {
```

Comment:
There are multiple other datatypes for single attributes https://www.rfc-editor.org/rfc/rfc7643.html#section-2.3. For core schema Attributes, it has been handled only for boolean attributes and all other attributes are considered as strings. This also needs to be handled properly by checking schema definitions.

You can handle the isCustomSchemaAttr case before this `else if` check. Handle the data type being string, boolena, integer .. etc   defined https://www.rfc-editor.org/rfc/rfc7643.html#section-2.3

Reply:
Resolved in https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/459/commits/a227f2e3af8b1ad2afaaab992d44c5f8e2b64c70